packages/react/src/ReactAct.js

```diff
@@ -91,21 +91,25 @@ export function act<T>(callback: () => T | Thenable<T>): Thenable<T> {
 
       if (__DEV__) {
         if (!didWarnNoAwaitAct && typeof Promise !== 'undefined') {
-          // eslint-disable-next-line no-undef
-          Promise.resolve()
-            .then(() => {})
-            .then(() => {
-              if (!wasAwaited) {
-                didWarnNoAwaitAct = true;
-                console.error(
-                  'You called act(async () => ...) without await. ' +
-                    'This could lead to unexpected testing behaviour, ' +
-                    'interleaving multiple act calls and mixing their ' +
-                    'scopes. ' +
-                    'You should - await act(async () => ...);',
-                );
-              }
-            });
+          if (result instanceof Promise) {
+            Promise.resolve(result)
```

Comment:
this should not be necessary - `instanceof Promise` is an antipattern, and `Promise.resolve()` works with any thenable or non-thenable.

If you want to determine if something *is* a native Promise, you'd do `Promise.resolve(result) === result`, but this doesn't cover generic thenables.

Reply:
Fixed. Nice catch, this was a leftover from the refactor I had overlooked.